packages/integrations/mdx/src/index.ts

```diff
@@ -1,17 +1,15 @@
 import { markdownConfigDefaults } from '@astrojs/markdown-remark';
-import { toRemarkInitializeAstroData } from '@astrojs/markdown-remark/dist/internal.js';
-import { compile as mdxCompile, type CompileOptions } from '@mdx-js/mdx';
+import { setAstroData } from '@astrojs/markdown-remark/dist/internal.js';
```

Comment:
Using `setAstroData` instead of `toRemarkInitializeAstroData` remark plugin helps avoid needing to re-create the processor everytime we render.

Reply:
Why not expose this from the package?

Reply:
It still feels a bit internal-ish so I put it in `internal.js` without thinking much. Need a bit of time to think about the ideal API, which I can also do though 😬

Reply:
Yeah, I think we should either expose it as is, or design a better API, but not use internal methods. I think it's ok if it's not the ideal API, we can change it later.

Reply:
Aight I updated to export `setVfileFrontmatter` instead from the main export, and removed `setAstroData` in the latest force push rebased to `main`.